### PR TITLE
Run unit tests with cake

### PR DESCRIPTION
**Pull request by tpluscode**
## Summary

With this PR anyone will be able to run unit tests locally through a powershell command

```ps
./build.ps1 -Target UnitTests
```

### VSTS setup

Cake runs the tests and produces the XML report. Here's a YAML which works fine for me to upload test results to the VSTS build job

```yaml
steps:
- task: cake-build.cake.cake-build-task.Cake@0
  displayName: 'Cake '
  inputs:
    target: UnitTests
  continueOnError: true

- task: PublishTestResults@2
  displayName: 'Publish Test Results TestResults/*.xml'
  inputs:
    testResultsFormat: VSTest
    testResultsFiles: 'TestResults/*.xml'
  condition: succeededOrFailed()
```

Note that the publish

### CODE REVIEW

**Comment by tpluscode** [comment]
Here's an en example build: https://dev.azure.com/tomaszpluskiewicz/test/_build/results?buildId=38&view=results

**Comment by sleupold** [comment]
@tpluscode 
can you please make sure, the instructions get included in DNN documentation (Wiki etc.) thanks

**Comment by tpluscode** [comment]
Sure @sleupold. Any guidance on where exactly to put those instructions? @ohine @mitchelsellers?

**Comment by ohine** [comment]
I've been working on a new blog post / wiki documentation page which will fully document all of our cake tasks. @tpluscode don't worry about getting this one task posted, I can easily include it in the page I'm working on.

File: Build/cake/unit-tests.cake [hunk 1]
```diff
@@ -0,0 +1,34 @@
+#tool "nuget:?package=Microsoft.TestPlatform&version=15.7.0"
+#tool "nuget:?package=NUnitTestAdapter&version=2.1.1"
+
+Task("EnsureAllProjectsBuilt")
+  .IsDependentOn("CompileSource")
```
Comment (ohine) [inline Build/cake/unit-tests.cake]: We can remove the dependent on `CompileSource` step since we are calling MSBuild manually inside of this step?

Comment (tpluscode) [reply Build/cake/unit-tests.cake]: Doesn't have to be `CompileSource` but we'd need at least 

```suggestion
  .IsDependentOn("UpdateDnnManifests")
  .IsDependentOn("Restore-NuGet-Packages")
```

to ensure that the unit tests target can be independently executed in a clean workspace

Comment (ohine) [reply Build/cake/unit-tests.cake]: That sounds better to me, I do not like the idea of compiling twice.

**Pull request merged**